.github/workflows/ci.yaml

```diff
@@ -0,0 +1,57 @@
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: "actions/checkout@v3"
+      - id: setup-haskell-cabal
+        uses: "haskell/actions/setup@v2"
+        with:
+          cabal-version: "${{ matrix.cabal }}"
+          enable-stack: false
+          ghc-version: "${{ matrix.ghc }}"
+      - uses: cachix/install-nix-action@v18
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - name: Update Hackage repository
+        run: cabal update
+      - name: cabal.project.local.ci
+        run: |
+          if [ -e cabal.project.local.ci ]; then
+            cp cabal.project.local.ci cabal.project.local
+          fi
+      - name: freeze
+        run: cabal freeze
+      - uses: "actions/cache@v3"
+        with:
+          key: "${{ runner.os }}-${{ matrix.ghc }}-cabal-${{ hashFiles('cabal.project.freeze') }}"
+          path: |
+            ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+            dist-newstyle
+      - name: Install dependencies
+        run: cabal build all --enable-tests --enable-benchmarks --only-dependencies
+      - name: build all
+        run: cabal build all --enable-tests --enable-benchmarks
+      - name: test all
+        run: cabal test all --enable-tests
+    strategy:
+      matrix:
+        cabal:
+          - '3.4'
+        ghc:
+          - '9.2.5'
+          - '9.0.2'
+          - '8.10.7'
+          - '8.8.4'
+  validate:
+    runs-on: ubuntu-latest
```

Comment:
you can use our GA runners with `self-hosted` tag that already has nix installed

Reply:
How can I use it? 
> error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

Reply:
You can either add reuse to the `devShell` and do something like `nix develop .#ci -c reuse lint` or extract reuse into a flake check, we have the corresponding [helper](https://github.com/serokell/serokell.nix/blob/master/overlay/default.nix#L41) in serokell.nix, usage example can be found [here](https://github.com/serokell/templates/blob/master/haskell.nix/flake.nix#L103)

Reply:
> file 'nixpkgs' was not found in the Nix search path 

This is done in order to avoid unexpected breaking changes in `<nixpkgs>`